app-frontend/src/app/pages/projects/edit/planetlabs/planetlabs.controller.js

```diff
@@ -0,0 +1,233 @@
+                        ],
+                        [
+                            -125.57373046874999,
+                            33.7243396617476
+                        ],
+                        [
+                            -117.26806640625,
+                            30.977609093348686
+                        ],
+                        [
+                            -112.74169921875,
+                            32.82421110161336
+                        ],
+                        [
+                            -113.203125,
+                            36.84446074079564
+                        ],
+                        [
+                            -112.8955078125,
+                            39.65645604812829
+                        ],
+                        [
+                            -112.2802734375,
+                            41.45919537950706
+                        ],
+                        [
+                            -116.05957031249999,
+                            42.47209690919285
+                        ],
+                        [
+                            -125.57373046874999,
+                            42.01665183556825
+                        ]
+                    ]
+                ]
+            }
+        };
+    }
+
+    getThumbnail(link) {
+        this.planetLabsService.getThumbnail(this.apiKey, link).then(
+            success => this.$log.log(success),
+            fail => this.$log.error(fail)
+        );
+    }
+
+    sendFilterRequest() {
+        let response =
+            this.planetLabsService.filterScenes(
+                this.apiKey,
+                JSON.stringify(this.filters));
+        response.then(
+            (success) => {
+                this.$log.log(success);
+                this.respondScenes = success.data;
+                if (success.data.features.length) {
+                    let thumbnailLink = this.respondScenes.features[0]._links.thumbnail;
+                    this.getThumbnail(thumbnailLink);
+                } else {
+                    this.$log.warn('No matching results.');
+                }
+            },
+            fail => this.$log.error(fail)
+        );
+    }
+
+    removeFilterConfig(filters, type, field) {
+        if (_.find(filters.filter.config, c => c.type === type && c.field_name === field)) {
+            _.remove(filters.filter.config, c => c.type === type && c.field_name === field);
+        }
+        return filters;
+    }
+
+    configureFilter(filters, type, field, leftVal, rightVal) {
+        let newFilters = this.removeFilterConfig(filters, type, field);
+        let configuredFilter = {
+            'type': type,
+            'field_name': field,
+            'config': {
+                'gte': leftVal,
+                'lte': rightVal
+            }
+        };
+        newFilters.filter.config.push(configuredFilter);
+        return newFilters;
+    }
+
+    setDataSource(dataSource) {
```

Comment:
May want to rename this `toggleDataSource` for clarity, but it's not critical if the code isn't long-term